cli/src/test/java/uk/ac/ebi/atlas/cli/experiment/CreateUpdateExperimentCommandTest.java

```diff
@@ -0,0 +1,129 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CreateUpdateExperimentCommandTest {
+
+    private CreateUpdateExperimentCommand createUpdateExperimentCommand;
+
+    private GxaExperimentCrud gxaExperimentCrudSpy;
+
+    @Mock
+    private ExperimentCrudDao experimentCrudDao;
+    @Mock
+    private ExperimentDesignFileWriterService experimentDesignFileWriterService;
+    @Mock
+    private ConfigurationTrader configurationTrader;
+    @Mock
+    private ExperimentChecker experimentChecker;
+    @Mock
+    private CondensedSdrfParser condensedSdrfParser;
+    @Mock
+    private IdfParser idfParser;
+
+    private List<UUID> randomUUIDs;
+
+    @Before
+    public void setUp() {
+        gxaExperimentCrudSpy = new GxaExperimentCrud(experimentCrudDao, experimentDesignFileWriterService,
+                configurationTrader, experimentChecker, condensedSdrfParser, idfParser);
+
+        randomUUIDs = generateRandomUUIDs(10);
+    }
+
+    @Test
+    public void whenNoAccessionPassed_ThenReturnCorrectExitCode() {
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(gxaExperimentCrudSpy);
+        var exitCode = createUpdateExperimentCommand.call();
+
+        final int exitCodeForNoExperimentAccessionSupplied = 1;
+        assertThat(exitCode).isEqualTo(exitCodeForNoExperimentAccessionSupplied);
+    }
+
+    @Test
+    public void createExperimentHasBeenCalledByNumberOfProvidedAccessions() {
+        GxaExperimentCrud spiedGxaExperimentCrud = spy(gxaExperimentCrudSpy);
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(spiedGxaExperimentCrud);
+        final UUID uuid1 = randomUUIDs.get(0);
+        final UUID uuid2 = randomUUIDs.get(1);
+        final UUID uuid3 = randomUUIDs.get(2);
+
+        doReturn(uuid1).doReturn(uuid2).doReturn(uuid3).
+                when(spiedGxaExperimentCrud).createExperiment(anyString(),eq(false));
+
+        CommandLine cmd = new CommandLine(createUpdateExperimentCommand);
+
+        int exitCode = cmd.execute("-e=acc1,acc2,acc3");
+
+        assertThat(exitCode).isEqualTo(0);
+        verify(spiedGxaExperimentCrud, times(3))
+                .createExperiment(anyString(), eq(false));
+    }
+
+    @Test
+    public void ifExperimentCreationFails_ThenRecordedInFailedAccessions() throws Exception{
+        GxaExperimentCrud spiedGxaExperimentCrud = spy(gxaExperimentCrudSpy);
+        createUpdateExperimentCommand = new CreateUpdateExperimentCommand(spiedGxaExperimentCrud);
+        final UUID uuid1 = randomUUIDs.get(0);
```

Comment:
You can `var`-ify these three declarations. Also notice that `var` implies `final`.

Reply:
I knew that I am going to left out some variable from your prefered `varified` state. Sorry about that. :-D